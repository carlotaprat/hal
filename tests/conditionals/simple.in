if True:
  print 1

if True:
   print 4

elif x:
  print 2

elif y:
  print 3

else:
  print 4
