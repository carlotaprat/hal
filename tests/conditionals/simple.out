digraph {

	ordering=out;
	ranksep=.4;
	bgcolor="lightgrey"; node [shape=box, fixedsize=false, fontsize=12, fontname="Helvetica-bold", fontcolor="blue"
		width=.25, height=.25, color="black", fillcolor="white", style="filled, solid, bold"];
	edge [arrowsize=.5, color="black", style="bold"]

  n0 [label="BLOCK"];
  n1 [label="IF_STMT"];
  n1 [label="IF_STMT"];
  n2 [label="FUNCALL"];
  n2 [label="FUNCALL"];
  n3 [label="True"];
  n4 [label="ARGS"];
  n5 [label="BLOCK"];
  n5 [label="BLOCK"];
  n6 [label="FUNCALL"];
  n6 [label="FUNCALL"];
  n7 [label="print"];
  n8 [label="ARGS"];
  n8 [label="ARGS"];
  n9 [label="1"];
  n10 [label="IF_STMT"];
  n10 [label="IF_STMT"];
  n11 [label="FUNCALL"];
  n11 [label="FUNCALL"];
  n12 [label="True"];
  n13 [label="ARGS"];
  n14 [label="BLOCK"];
  n14 [label="BLOCK"];
  n15 [label="FUNCALL"];
  n15 [label="FUNCALL"];
  n16 [label="print"];
  n17 [label="ARGS"];
  n17 [label="ARGS"];
  n18 [label="4"];
  n19 [label="BLOCK"];
  n19 [label="BLOCK"];
  n20 [label="IF_STMT"];
  n20 [label="IF_STMT"];
  n21 [label="FUNCALL"];
  n21 [label="FUNCALL"];
  n22 [label="x"];
  n23 [label="ARGS"];
  n24 [label="BLOCK"];
  n24 [label="BLOCK"];
  n25 [label="FUNCALL"];
  n25 [label="FUNCALL"];
  n26 [label="print"];
  n27 [label="ARGS"];
  n27 [label="ARGS"];
  n28 [label="2"];
  n29 [label="BLOCK"];
  n29 [label="BLOCK"];
  n30 [label="IF_STMT"];
  n30 [label="IF_STMT"];
  n31 [label="FUNCALL"];
  n31 [label="FUNCALL"];
  n32 [label="y"];
  n33 [label="ARGS"];
  n34 [label="BLOCK"];
  n34 [label="BLOCK"];
  n35 [label="FUNCALL"];
  n35 [label="FUNCALL"];
  n36 [label="print"];
  n37 [label="ARGS"];
  n37 [label="ARGS"];
  n38 [label="3"];
  n39 [label="BLOCK"];
  n39 [label="BLOCK"];
  n40 [label="FUNCALL"];
  n40 [label="FUNCALL"];
  n41 [label="print"];
  n42 [label="ARGS"];
  n42 [label="ARGS"];
  n43 [label="4"];

  n0 -> n1 // "BLOCK" -> "IF_STMT"
  n1 -> n2 // "IF_STMT" -> "FUNCALL"
  n2 -> n3 // "FUNCALL" -> "True"
  n2 -> n4 // "FUNCALL" -> "ARGS"
  n1 -> n5 // "IF_STMT" -> "BLOCK"
  n5 -> n6 // "BLOCK" -> "FUNCALL"
  n6 -> n7 // "FUNCALL" -> "print"
  n6 -> n8 // "FUNCALL" -> "ARGS"
  n8 -> n9 // "ARGS" -> "1"
  n0 -> n10 // "BLOCK" -> "IF_STMT"
  n10 -> n11 // "IF_STMT" -> "FUNCALL"
  n11 -> n12 // "FUNCALL" -> "True"
  n11 -> n13 // "FUNCALL" -> "ARGS"
  n10 -> n14 // "IF_STMT" -> "BLOCK"
  n14 -> n15 // "BLOCK" -> "FUNCALL"
  n15 -> n16 // "FUNCALL" -> "print"
  n15 -> n17 // "FUNCALL" -> "ARGS"
  n17 -> n18 // "ARGS" -> "4"
  n10 -> n19 // "IF_STMT" -> "BLOCK"
  n19 -> n20 // "BLOCK" -> "IF_STMT"
  n20 -> n21 // "IF_STMT" -> "FUNCALL"
  n21 -> n22 // "FUNCALL" -> "x"
  n21 -> n23 // "FUNCALL" -> "ARGS"
  n20 -> n24 // "IF_STMT" -> "BLOCK"
  n24 -> n25 // "BLOCK" -> "FUNCALL"
  n25 -> n26 // "FUNCALL" -> "print"
  n25 -> n27 // "FUNCALL" -> "ARGS"
  n27 -> n28 // "ARGS" -> "2"
  n20 -> n29 // "IF_STMT" -> "BLOCK"
  n29 -> n30 // "BLOCK" -> "IF_STMT"
  n30 -> n31 // "IF_STMT" -> "FUNCALL"
  n31 -> n32 // "FUNCALL" -> "y"
  n31 -> n33 // "FUNCALL" -> "ARGS"
  n30 -> n34 // "IF_STMT" -> "BLOCK"
  n34 -> n35 // "BLOCK" -> "FUNCALL"
  n35 -> n36 // "FUNCALL" -> "print"
  n35 -> n37 // "FUNCALL" -> "ARGS"
  n37 -> n38 // "ARGS" -> "3"
  n30 -> n39 // "IF_STMT" -> "BLOCK"
  n39 -> n40 // "BLOCK" -> "FUNCALL"
  n40 -> n41 // "FUNCALL" -> "print"
  n40 -> n42 // "FUNCALL" -> "ARGS"
  n42 -> n43 // "ARGS" -> "4"

}

